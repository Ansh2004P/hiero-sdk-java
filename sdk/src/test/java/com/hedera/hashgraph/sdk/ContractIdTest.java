package com.hedera.hashgraph.sdk;

import io.github.jsonSnapshot.SnapshotMatcher;

import com.google.protobuf.InvalidProtocolBufferException;

import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class ContractIdTest {
    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void fromString() {
        SnapshotMatcher.expect(ContractId.fromString("0.0.5005").toProtobuf().toString()).toMatchSnapshot();
    }

    @Test
    void fromSolidityAddress() {
        SnapshotMatcher.expect(ContractId.fromSolidityAddress("000000000000000000000000000000000000138D").toProtobuf().toString()).toMatchSnapshot();
    }

    @Test
    void fromSolidityAddressWith0x() {
        SnapshotMatcher.expect(ContractId.fromSolidityAddress("0x000000000000000000000000000000000000138D").toProtobuf().toString()).toMatchSnapshot();
    }

    @Test
    void toBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(new ContractId(5005).toBytes()).toMatchSnapshot();
    }

    @Test
    void fromBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(ContractId.fromBytes(new ContractId(5005).toBytes())).toMatchSnapshot();
    }

    @Test
    void toSolidityAddress() {
        SnapshotMatcher.expect(new ContractId(5005).toSolidityAddress()).toMatchSnapshot();
    }
}
