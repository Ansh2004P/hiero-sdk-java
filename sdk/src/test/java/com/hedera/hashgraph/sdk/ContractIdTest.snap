com.hedera.hashgraph.sdk.ContractIdTest.fromBytes=[
  {
    "shard": 0,
    "realm": 0,
    "num": 5005
  }
]


com.hedera.hashgraph.sdk.ContractIdTest.fromSolidityAddress=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.fromSolidityAddressWith0x=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.fromString=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.shouldSerialize=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.shouldSerializeFromSolidityAddress=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.shouldSerializeFromSolidityAddressWith0x=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.shouldSerializeFromString=[
  "# com.hedera.hashgraph.sdk.proto.ContractID@bc8a0\ncontract_num: 5005\nrealm_num: 0\nshard_num: 0"
]


com.hedera.hashgraph.sdk.ContractIdTest.toBytes=[
  "GI0n"
]


com.hedera.hashgraph.sdk.ContractIdTest.toSolidityAddress=[
  "000000000000000000000000000000000000138d"
]